File: packages/evals/tasks/bench/act/vantechjournal.ts
import { defineBenchTask } from "../../../framework/defineTask.js";

export default defineBenchTask(
  { name: "vantechjournal" },
  async ({ debugUrl, sessionUrl, stagehand, page, logger }) => {
    try {
      await page.goto("https://vantechjournal.com");

      await stagehand.act("click on page 'recommendations'");

      const expectedUrl = "https://vantechjournal.com/recommendations";
      const currentUrl = await page.url();

      return {
        _success: currentUrl === expectedUrl,
        currentUrl,
        expectedUrl,
        debugUrl,
        sessionUrl,
        logs: logger.getLogs(),
      };
    } catch (error) {
      return {
        _success: false,
        error: String(error),
        debugUrl,
        sessionUrl,
        logs: logger.getLogs(),
      };
    }
  },
);

File: packages/sdk-ruby/evals/tasks/act/vantechjournal.rb
# frozen_string_literal: true

# Port of packages/evals/tasks/bench/act/vantechjournal.ts

Evals.define_task("vantechjournal") do |t|
  t.page.goto("https://vantechjournal.com")

  t.stagehand.act("click on page 'recommendations'")

  expected_url = "https://vantechjournal.com/recommendations"
  current_url = t.page.url

  { _success: current_url == expected_url, currentUrl: current_url, expectedUrl: expected_url }
end
